api/src/convert/block_contents.rs

```diff
@@ -28,15 +41,37 @@ impl TryFrom<&blockchain::BlockContents> for mc_transaction_core::BlockContents
     type Error = ConversionError;
 
     fn try_from(source: &blockchain::BlockContents) -> Result<Self, Self::Error> {
-        let mut key_images: Vec<KeyImage> = Vec::new();
-        for key_image in source.get_key_images() {
-            key_images.push(KeyImage::try_from(key_image)?);
-        }
-
-        let mut outputs: Vec<tx::TxOut> = Vec::new();
-        for output in source.get_outputs() {
-            outputs.push(tx::TxOut::try_from(output)?);
-        }
-        Ok(BlockContents::new(key_images, outputs))
+        let key_images: Vec<KeyImage> = source
+            .get_key_images()
+            .iter()
+            .map(KeyImage::try_from)
+            .collect::<Result<_, _>>()?;
+
+        let outputs: Vec<TxOut> = source
+            .get_outputs()
+            .iter()
+            .map(TxOut::try_from)
+            .collect::<Result<_, _>>()?;
```

Comment:
Optional/Equivalent: Move the type declaration into the collect call
```suggestion
        let key_images = source
            .get_key_images()
            .iter()
            .map(KeyImage::try_from)
            .collect::<Result<Vec<_>, _>>()?;

        let outputs = source
            .get_outputs()
            .iter()
            .map(TxOut::try_from)
            .collect::<Result<Vec<_>, _>>()?;
```